packages/react-mutation-mapper/src/component/track/Track.tsx

```diff
@@ -198,40 +231,71 @@ export default class Track extends React.Component<TrackProps, {}> {
     }
 
     get items() {
-        this.circles = {};
+        this.shapes = {};
 
         return (this.props.trackItems || []).map((spec, index) => {
-            return (
-                <TrackCircle
-                    ref={(circle: TrackCircle) => {
-                        if (circle !== null) {
-                            this.circles[index] = circle;
+            if (spec.itemType === TrackItemType.RECTANGLE) {
+                return (
+                    <TrackItem
+                        ref={(rect: TrackItem) => {
+                            if (rect !== null) {
+                                this.shapes[index] = rect;
+                            }
+                        }}
+                        key={spec.startCodon}
+                        hitZoneClassName={`${this.props.idClassPrefix}${index}`}
+                        hitZoneXOffset={this.props.xOffset}
+                        x={
+                            (spec.startCodon / this.props.proteinLength) *
+                            this.props.width
                         }
-                    }}
-                    key={spec.codon}
-                    hitZoneClassName={`${this.props.idClassPrefix}${index}`}
-                    hitZoneXOffset={this.props.xOffset}
-                    x={
-                        (this.props.width / this.props.proteinLength) *
-                        spec.codon
-                    }
-                    y={this.svgHeight / 2}
-                    radius={
-                        this.props.dataStore.isPositionSelected(spec.codon) ||
-                        this.props.dataStore.isPositionHighlighted(spec.codon)
-                            ? 5
-                            : 2.8
-                    }
-                    hoverRadius={5}
-                    spec={spec}
-                />
-            );
+                        y={this.svgHeight / 2}
+                        dim1={
+                            ((spec.endCodon! - spec.startCodon) /
+                                this.props.proteinLength) *
+                            this.props.width
+                        }
+                        dim2={50}
+                        spec={{ ...spec, endCodon: spec.endCodon! }}
+                    />
+                );
+            } else {
```

Comment:
In both cases we create a `TrackItem`. It might be better to handle the shape logic within `TrackItem` implementation.